src/containers/Repos/index.tsx

```diff
@@ -8,68 +8,84 @@ import { useFetchRecommendationQuery } from "@features/repos/api/getRecommendati
 import { useRouter } from "next/router";
 import { Trans } from "@lingui/macro";
 import { skipToken } from "@reduxjs/toolkit/query";
+import styled from "@emotion/styled";
+import Link from "next/link";
 
 interface RepoContainerProps {
   padding?: number;
   maxwidth?: number;
 }
 
+const StyledSpan = styled.span`
+  color: darkblue;
+`;
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+`;
 const Repos: React.FC<RepoContainerProps> = ({ maxwidth }) => {
   const router = useRouter();
-  const [repoName, setRepoName] = useState<string>("react");
+  const [repoName, setRepoName] = useState<string>("");
   const [page, setPage] = useState<number>(1);
 
   const { data, error, isLoading, isFetching } = useFetchRecommendationQuery(
     isEmpty(repoName)
       ? skipToken
       : {
-          repoName,
-          page: Number(page),
-        },
+        repoName,
+        page,
+      },
     { skip: router.isFallback }
   );
 
   const handlePageChange = (_: React.ChangeEvent<unknown>, pageNumber: number) => {
     setPage(pageNumber);
-    router.push(`/?search=${repoName}&page=${pageNumber}`, undefined, {
-      shallow: true,
-      scroll: true,
-    });
+    updateUrlParams(repoName, pageNumber);
   };
 
   const handleRepoSearch = debounce((repoName: string) => {
     setRepoName(repoName);
     setPage(1);
+    updateUrlParams(repoName, 1); // Update URL params immediately
   }, 500);
+  const updateUrlParams = (repoName: string, pageNumber: number) => {
+    router.push(`/?search=${repoName}&page=${pageNumber}`, undefined, {
+      shallow: true,
+      scroll: true,
+    });
+  };
 
   useEffect(() => {
     if (router.isReady) {
-      setRepoName(router.query?.search as string);
-      setPage(router.query?.page as unknown as number);
+      const searchParam = router.query?.search as string;
+      setRepoName(searchParam || "");
+      setPage(Number(router.query?.page) || 1);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can we remove the eslint-disable-next-line? doesn't seem like we need it neccessarily here

Reply:
removed